src/display/utils.cc

```diff
@@ -257,21 +257,62 @@ print_client_version(char* first, char* last, const torrent::ClientInfo& clientI
   }
 }
 
+int64_t
+retrieve_throttle_up_value(const torrent::Object::string_type& name, bool rate) {
+  core::ThrottleMap::iterator itr = control->core()->throttles().find(name);
+
+  if (itr == control->core()->throttles().end()) {
+    return (int64_t)-1;
+  } else {
+    torrent::Throttle* throttle = itr->second.first;
+    int64_t throttle_max = (int64_t)throttle->max_rate();
+    if (rate) {
+      if (throttle_max > 0)
+	return (int64_t)throttle->rate()->rate();
+      else
+	return (int64_t)-1;
+    } else {
+      return throttle_max;
+    }
+  }
+}
+
 char*
 print_status_info(char* first, char* last) {
-  if (!torrent::up_throttle_global()->is_throttled())
+  std::string throttle_up_name = rpc::call_command_string("ui.status.throttle_up_name").c_str();
```

Comment:
Move this inside the 'else' block so you only initialize the variable right before you use it, this isn't C.

Also, no need to use 'c_str' conversion here, just use the std::string it returns.